File: backend/app/schemas/ai_run.py
"""Schemas for AIRun."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.ai_run import AIRunStatus, AIRunType


class AIRunResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    run_type: AIRunType
    resource_type: str
    resource_id: str
    status: AIRunStatus
    model: str | None
    error_code: str | None
    user_message: str | None
    created_at: datetime
    started_at: datetime | None
    completed_at: datetime | None

File: frontend/types/ai_run.ts
export type AIRunType =
  | "PROGRESS_ANALYSIS"
  | "ISSUE_ADVICE"
  | "DAILY_SUMMARY"
  | "CONVERSATION_SUMMARY";

export type AIRunStatus = "QUEUED" | "RUNNING" | "SUCCEEDED" | "FAILED" | "DISABLED";

export interface AIRun {
  id: number;
  run_type: AIRunType;
  resource_type: string;
  resource_id: string;
  status: AIRunStatus;
  model: string | null;
  error_code: string | null;
  user_message: string | null;
  created_at: string;
  started_at: string | null;
  completed_at: string | null;
}
